src/main/kotlin/org/rust/lang/core/completion/RsLifetimeCompletionProvider.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.ide.refactoring.RsNamesValidator
+import org.rust.lang.core.RsPsiPattern
+
+object RsLifetimeCompletionProvider : RsCompletionProvider() {
+
+    override val elementPattern: ElementPattern<PsiElement> get() = RsPsiPattern.lifetimeIdentifier
+
+    override fun addCompletions(
+        parameters: CompletionParameters,
+        context: ProcessingContext,
+        result: CompletionResultSet
+    ) {
+        for (lifetime in RsNamesValidator.RESERVED_LIFETIME_NAMES.sorted()) {
```

Comment:
Are you sure that we want to suggest `'_` at all?
It doesn't look such common case as `static`, at least from my point of view but will always appear in completion list. Also, `_` is just a single symbol but `static` a quite long word

Reply:
Maybe it could be useful for learning/discoverability? To let people know that they can even use `'_`.

Reply:
I don't think that's necessary. There are already lifetime elision lints & intentions which achieve that goal. It also feels natural to try `_` as the lifetime name, since it is allowed in place of type generics.